utils/json-utils/src/main/scala/pl/touk/nussknacker/engine/json/swagger/extractor/JsonToNuStruct.scala

```diff
@@ -59,8 +59,8 @@ object JsonToNuStruct {
         extract[JsonNumber](_.asNumber, _.toBigDecimal.map(_.bigDecimal).orNull)
       case SwaggerArray(elementType) =>
         extract[Vector[Json]](_.asArray, _.zipWithIndex.map { case (el, idx) => JsonToNuStruct(el, elementType, s"$path[$idx]") }.asJava)
```

Comment:
Above, exactly at 32 line you have to change the conditions:
```
case (key, value) if elementType.contains(key) 
```
Because right now struct will cut additional fields..

Reply:
it is expected acctualy, please see https://github.com/TouK/nussknacker/pull/3709/files#diff-bc73474afc18bd51d0dd81754c63dd2b308b8e457ed1d5bac55a98d11b834777R127

we handle `addionalFields` only when used without `properties` -> and in that case such schema is typed to Map, not Object.

Reply:
Right, great! :)